ADALiOS/ADALiOS/ADAuthenticationWebViewController.m

```diff
@@ -153,12 +161,27 @@ - (BOOL)webView:(UIWebView *)webView shouldStartLoadWithRequest:(NSURLRequest *)
         // This event is explicitly scheduled on the main thread as it is UI related.
         NSAssert( nil != _delegate, @"Delegate object was lost" );
         
-        dispatch_async( dispatch_get_main_queue(), ^{ [_delegate webAuthenticationDidCompleteWithURL:request.URL]; } );
+        NSURL* url = request.URL;
+        
+        dispatch_async( dispatch_get_main_queue(), ^{ [_delegate webAuthenticationDidCompleteWithURL:url]; } );
         
         // Tell the web view that this URL should not be loaded.
         return NO;
     }
     
+#if AD_BROKER
+    if(![request.allHTTPHeaderFields valueForKey:pKeyAuthHeader]){
+        // Create a mutable copy of the immutable request and add more headers
+        NSMutableURLRequest *mutableRequest = [request mutableCopy];
+        [mutableRequest addValue:pKeyAuthHeaderVersion forHTTPHeaderField:pKeyAuthHeader];
+        
+        // Now set our request variable with an (immutable) copy of the altered request
+        request = [mutableRequest copy];
+        [webView loadRequest:request];
+        return NO;
+    }
+#endif // AD_BROKER
+    
```

Comment:
sorry. my bad. this needs to die. remove this if block as this was moved to [ADCustomHeaderHandler]